rocketpy/Function.py

```diff
@@ -473,6 +473,160 @@ def setDiscrete(
+        """This method transforms a Function instance defined from callables into a
+        Function instance defined by a list of discrete points.
+        It does so based on a model Function, from which it retrieves the domain,
+        domain name, interpolation method and extrapolation method.
+        It then evaluates the original Function instance in all points of the retrieved
+        domain to generate the list of discrete points that will be used for
+        interpolation when this Function is called.
+
+        Parameters
+        ----------
+        modelFunction : Function
+            Function object that will be used to define the sampling points,
+            interpolation method and extrapolation method.
+            Must be a Function whose source attribute is a list (i.e. a list based
+            Function instance).
+            Must have the same domain dimension as the Function to be discretized.
+
+        oneByOne : boolean, optional
+            If True, evaluate Function in each sample point separately. If
+            False, evaluates Function in vectorized form. Default is True.
+
+        Returns
+        -------
+        self : Function
+
+        See also
+        --------
+        Function.setDiscrete
+
+        Examples
+        --------
+        This method is particularly useful when algebraic operations is carried out
+        using Function instances defined by different discretized domains (same range,
+        but different mesh size). Once an algebraic operation is done, it will not
+        directly be applied between the list of discrete points of the two Function
+        instances. Instead, the result will be a Function instance defined by a callable
+        that calls both Function instances and performs the operation. This makes the
+        evaluation of the resulting Function inefficient, due to extra function calling
+        overhead and multiple interpolations being carried out.
+
+        >>> from rocketpy import Function
+        >>> f = Function([(0, 0), (1, 1), (2, 4), (3, 9), (4, 16)])
+        >>> g = Function([(0, 0), (2, 2), (4, 4)])
+        >>> h = f * g
+        >>> type(h.source)
+        <class 'function'>
+
+        Therefore, it is good practice to make sure both Function instances are defined
+        by the same domain, i.e. by the same list of mesh points. This way, the
+        algebraic operation will be carried out directly between the lists of discrete
+        points, generating a new Function instance defined by this result. When it is
+        evaluated, there are no extra function calling overheads neither multiple
+        interpolations.
+
+        >>> g.setDiscreteBasedOnModel(f)
+        Function from R1 to R1 : (Scalar) → (Scalar)
+        >>> h = f * g
+        >>> h.source
+        array([[ 0.,  0.],
+               [ 1.,  1.],
+               [ 2.,  8.],
+               [ 3., 27.],
+               [ 4., 64.]])
```

Comment:
I do believe there is good use of not converting both Functions to the same domain before algebraic operations are carried out. I you will evaluate the result in just a few points, this is more efficient then using the `setDiscreteBasedOnModel`.

Since the current implementation allow for both options (converting the domain and not converting), I prefer to keep it like this for now.